### PR TITLE
Brings back crew transfer

### PR DESCRIPTION
**Pull request by Alberyk**
This pr adds another way to end the round: crew transfer

Since the horizon has no shuttle, it works like a bluespace jump without the blue overlay.

The crew can now start a bluespace jump instead of an evacuation as well.

It implements the suggestion from this pr: https://forums.aurorastation.org/topic/17710-make-bluespace-jumping-mechanically-supportedround-ending

### CODE REVIEW

**Comment by TheDanseMacabre** [comment]
Might be better to call it "shift change"/"crew change" instead of "crew transfer". The latter implies the crew leaving the ship, if you ask me.

File: code/controllers/subsystems/evacuation/evacuation_shuttle.dm [hunk 1]
```diff
@@ -54,12 +58,12 @@
 
 	. = ..()
 	// Arm the escape pods.
-	if (emergency_evacuation)
+	if (evacuation_type == "emergency")
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/evacuation/evacuation_shuttle.dm:61]: these should be defines, but other than that the PR is good to go

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

File: code/game/machinery/case_button.dm [hunk 2]
```diff
@@ -100,14 +100,30 @@
...
+	else
+		evacuation_type = TRANSFER_EMERGENCY
+
+/obj/machinery/case_button/shuttle/examine(mob/user)
+	..(user)
+	switch(evacuation_type)
+		if(TRANSFER_EMERGENCY)
+			to_chat(user, "The button is set to start an emergency evacuation.")
+		if(TRANSFER_JUMP)
+			to_chat(user, "The button is set to start a bluespace jump.")
```
Comment (Arrow768) [inline code/game/machinery/case_button.dm:122]: Repurposing the button is imho not very intuitive and will lead to situations where the wrong action is performed.

As you suggested on discord, moving the evacuation to the swiper is a better option.

**Review by Arrow768 — CHANGES_REQUESTED** [review]

**Review by Arrow768 — APPROVED** [review]

File: code/controllers/subsystems/evacuation/evacuation_pods.dm [hunk 3]
```diff
@@ -15,14 +17,16 @@
...
-		EVAC_OPT_CANCEL_BLUESPACE_JUMP = new /datum/evacuation_option/cancel_bluespace_jump()
+		EVAC_OPT_CANCEL_BLUESPACE_JUMP = new /datum/evacuation_option/cancel_bluespace_jump(),
+		EVAC_OPT_CANCEL_CREW_TRANSFER = new /datum/evacuation_option/cancel_crew_transfer()
 	)
 
 /datum/evacuation_controller/starship/finish_preparing_evac()
 	. = ..()
 	// Arm the escape pods.
-	if (emergency_evacuation)
+	if (evacuation_type == TRANSFER_EMERGENCY)
```
Comment (SleepyGemmy) [inline code/controllers/subsystems/evacuation/evacuation_pods.dm:29]:
```suggestion
	if(evacuation_type == TRANSFER_EMERGENCY)
```

File: code/controllers/subsystems/evacuation/evacuation_pods.dm [hunk 4]
```diff
@@ -31,22 +35,25 @@
...
-		priority_announcement.Announce(replacetext(replacetext(current_map.emergency_shuttle_leaving_dock, "%dock_name%", "[current_map.dock_name]"),  "%ETA%", "[round(get_eta()/60,1)] minute\s"))
-	else
-		// Bluespace Jump
-		priority_announcement.Announce(replacetext(replacetext(current_map.shuttle_leaving_dock, "%dock_name%", "[current_map.dock_name]"),  "%ETA%", "[round(get_eta()/60,1)] minute\s"))
-		SetUniversalState(/datum/universal_state/bluespace_jump, arguments=list(current_map.station_levels))
+	switch(evacuation_type)
+		if ("evacuation")
+			// Abandon Ship
+			for (var/datum/shuttle/autodock/ferry/escape_pod/pod in escape_pods) // Launch the pods!
+				if (!pod.arming_controller || pod.arming_controller.armed)
```
Comment (SleepyGemmy) [inline code/controllers/subsystems/evacuation/evacuation_pods.dm:42]:
```suggestion
		if("evacuation")
			// Abandon Ship
			for(var/datum/shuttle/autodock/ferry/escape_pod/pod in escape_pods) // Launch the pods!
				if(!pod.arming_controller || pod.arming_controller.armed)
```

File: code/controllers/subsystems/evacuation/evacuation_pods.dm [hunk 5]
```diff
@@ -55,10 +62,13 @@
 	if (is_idle())
 		return list(evacuation_options[EVAC_OPT_BLUESPACE_JUMP], evacuation_options[EVAC_OPT_ABANDON_SHIP])
 	if (is_evacuating())
-		if (emergency_evacuation)
-			return list(evacuation_options[EVAC_OPT_CANCEL_ABANDON_SHIP])
-		else
-			return list(evacuation_options[EVAC_OPT_CANCEL_BLUESPACE_JUMP])
+		switch(evacuation_type)
+			if (TRANSFER_EMERGENCY)
```
Comment (SleepyGemmy) [inline code/controllers/subsystems/evacuation/evacuation_pods.dm:66]:
```suggestion
			if(TRANSFER_EMERGENCY)
```

File: code/controllers/subsystems/evacuation/evacuation_pods.dm [hunk 6]
```diff
@@ -105,6 +115,28 @@
...
+	if (evacuation_controller.deny)
+		to_chat(user, "Unable to initiate crew transfer preparation.")
+		return
+	if (evacuation_controller.is_on_cooldown())
+		to_chat(user, evacuation_controller.get_cooldown_message())
+		return
+	if (evacuation_controller.is_evacuating())
+		to_chat(user, "Crew transfer preparation already in progress.")
+		return
+	if (evacuation_controller.call_evacuation(user, 0))
```
Comment (SleepyGemmy) [inline code/controllers/subsystems/evacuation/evacuation_pods.dm:137]:
```suggestion
	if(!evacuation_controller)
		return
	if(evacuation_controller.deny)
		to_chat(user, "Unable to initiate crew transfer preparation.")
		return
	if(evacuation_controller.is_on_cooldown())
		to_chat(user, evacuation_controller.get_cooldown_message())
		return
	if(evacuation_controller.is_evacuating())
		to_chat(user, "Crew transfer preparation already in progress.")
		return
	if(evacuation_controller.call_evacuation(user, 0))
```

File: code/controllers/subsystems/evacuation/evacuation_pods.dm [hunk 7]
```diff
@@ -127,6 +159,17 @@
...
 
+/datum/evacuation_option/cancel_crew_transfer
+	option_text = "Cancel crew transfer"
+	option_desc = "cancel the crew transfer preparation"
+	option_target = EVAC_OPT_CANCEL_BLUESPACE_JUMP
+	needs_syscontrol = TRUE
+	silicon_allowed = FALSE
+
+/datum/evacuation_option/cancel_crew_transfer/execute(mob/user)
+	if (evacuation_controller && evacuation_controller.cancel_evacuation())
```
Comment (SleepyGemmy) [inline code/controllers/subsystems/evacuation/evacuation_pods.dm:170]:
```suggestion
	if(evacuation_controller && evacuation_controller.cancel_evacuation())
```

File: code/controllers/subsystems/evacuation/evacuation_shuttle.dm [hunk 8]
```diff
@@ -41,10 +41,14 @@
 
 	state = EVAC_IN_TRANSIT
 
-	if (emergency_evacuation)
-		priority_announcement.Announce(replacetext(replacetext(current_map.emergency_shuttle_leaving_dock, "%dock%", "[current_map.dock_name]"),  "%ETA%", "[round(get_eta()/60,1)] minute\s"))
-	else
-		priority_announcement.Announce(replacetext(replacetext(current_map.shuttle_leaving_dock, "%dock%", "[current_map.dock_name]"),  "%ETA%", "[round(get_eta()/60,1)] minute\s"))
+	switch(evacuation_type)
+		if (TRANSFER_EMERGENCY)
```
Comment (SleepyGemmy) [inline code/controllers/subsystems/evacuation/evacuation_shuttle.dm:45]:
```suggestion
		if(TRANSFER_EMERGENCY)
```

File: code/controllers/subsystems/evacuation/evacuation_shuttle.dm [hunk 9]
```diff
@@ -54,12 +58,12 @@
 
 	. = ..()
 	// Arm the escape pods.
-	if (emergency_evacuation)
+	if (evacuation_type == TRANSFER_EMERGENCY)
```
Comment (SleepyGemmy) [inline code/controllers/subsystems/evacuation/evacuation_shuttle.dm:61]:
```suggestion
	if(evacuation_type == TRANSFER_EMERGENCY)
```

File: code/datums/discord/webhook.dm [hunk 10]
```diff
@@ -68,6 +68,12 @@ var/list/global_webhooks = list()
 /proc/post_webhook_event(var/tag, var/list/data)
 	set background = 1
 
+	var/escape_text
+	if (evacuation_controller.evacuation_type == TRANSFER_EMERGENCY)
```
Comment (SleepyGemmy) [inline code/datums/discord/webhook.dm:72]:
```suggestion
	if(evacuation_controller.evacuation_type == TRANSFER_EMERGENCY)
```

File: code/datums/late_choices.dm [hunk 11]
```diff
@@ -67,10 +67,10 @@
 		if(evacuation_controller.has_evacuated()) //Shuttle is going to centcomm, not recalled
 			shuttle_status = "post-evac"
 		if(evacuation_controller.is_evacuating())
-			if (evacuation_controller.emergency_evacuation) // Emergency shuttle is past the point of no recall
+			if (evacuation_controller.evacuation_type == TRANSFER_EMERGENCY) // Emergency shuttle is past the point of no recall
```
Comment (SleepyGemmy) [inline code/datums/late_choices.dm:70]:
```suggestion
			if(evacuation_controller.evacuation_type == TRANSFER_EMERGENCY) // Emergency shuttle is past the point of no recall
```

File: code/game/gamemodes/game_mode.dm [hunk 12]
```diff
@@ -388,12 +388,17 @@ var/global/list/additional_antag_types = list()
 				ghosts++
 
 	var/text = ""
+	var/escape_text
+	if (evacuation_controller.evacuation_type == TRANSFER_EMERGENCY)
```
Comment (SleepyGemmy) [inline code/game/gamemodes/game_mode.dm:392]:
```suggestion
	if(evacuation_controller.evacuation_type == TRANSFER_EMERGENCY)
```

File: code/game/machinery/case_button.dm [hunk 13]
```diff
@@ -98,7 +98,7 @@
 
 
 /obj/machinery/case_button/shuttle
-	name = "\improper Emergency Shuttle Button"
+	name = "\improper Bluespace Jump Button"
```
Comment (SleepyGemmy) [inline code/game/machinery/case_button.dm:101]:
```suggestion
	name = "bluespace jump button"
```

File: code/modules/admin/verbs/randomverbs.dm [hunk 14]
```diff
@@ -907,8 +907,8 @@ Traitors and the like can also be revived with the previous role mostly intact.
 		if(input("The evacuation will just be cancelled if you call it. Call anyway?") in list("Confirm", "Cancel") != "Confirm")
 			return
 
-	var/choice = input("Is this an emergency evacuation or a crew transfer?") in list("Emergency", "Crew Transfer")
-	evacuation_controller.call_evacuation(usr, (choice == "Emergency"))
+	var/choice = input("Is this an emergency evacuation, bluespace jump or a crew transfer?") in list(TRANSFER_EMERGENCY, TRANSFER_CREW, TRANSFER_JUMP)
```
Comment (SleepyGemmy) [inline code/modules/admin/verbs/randomverbs.dm:910]:
```suggestion
	var/choice = input("Is this an emergency evacuation, bluespace jump, or a crew transfer?") in list(TRANSFER_EMERGENCY, TRANSFER_CREW, TRANSFER_JUMP)
```

**Pull request merged**